lib/parsing.js

```diff
@@ -1,5 +1,30 @@
 'use strict';
 
+/**
+ * @typedef AstNode
+ * @property {string} type
+ * @property {'none'|'single'|'double'} quoteStyle?
```

Comment:
`tsc` has no problem with this style at least, don't remember any reference on the top of my head that I can point to though

Reply:
Pretty sure that's because it stops parsing and treats `?` as the documentation for the property. I tested it and it's definitely not treated as optional. Can you switch to the `[quoteStyle]` syntax instead?

Reply:
Done @sandersn 🙂